tests/virtual_drive.py

```diff
@@ -0,0 +1,118 @@
+import os
+import socket
+import time
+from enum import Enum
+from threading import Thread
+
+import xml.etree.ElementTree as ET
+
+from ingenialink.constants import ETH_BUF_SIZE
+from ingenialink.utils.mcb import MCB
+
+class MSG_TYPE(Enum):
+    RECEIVED = "RECEIVED"
+    SENT = "SENT"
+
+
+class VirtualDrive(Thread):
+    ACK_CMD = 3
+    WRITE_CMD = 2
+    READ_CMD = 1
+
+    def __init__(self, ip, port, config_file="./tests/resources/virtual_drive.xcf"):
+        super(VirtualDrive, self).__init__()
+        self.ip = ip
+        self.port = port
+        self.config_file = config_file
+        self.socket = None
+        self.__stop = False
+        self.device_info = None
+        self.registers = {}
+        self.__logger = []
+        self._load_configuration_file()
+
+    def run(self):
+        ''' Open socket and listen messages '''
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        server_address = (self.ip, self.port)
+        self.socket.bind(server_address)
+        self.socket.settimeout(2)
+        while not self.__stop:
+            if self.socket is not None:
```

Comment:
This check will not be needed if you move the creation of the socket to the constructor.